modules/search/instant-search/components/search-app.jsx

```diff
@@ -35,23 +36,21 @@ class SearchApp extends Component {
 		this.props.resultFormat = 'minimal';
 		this.props.aggregations = buildFilterAggregations( this.props.options.widgets );
 		this.props.widgets = this.props.options.widgets ? this.props.options.widgets : [];
+		this.isSearchPage = this.props.initialValue !== '';
 		this.state = {
 			query: this.props.initialValue,
 			sort: this.props.initialSort,
 			results: {},
 			loading: false,
 		};
-		this.getResults = debounce( this.getResults, 200 );
+		this.getDebouncedResults = debounce( this.getResults, 200 );
 		this.getResults( this.state.query, getFilterQuery(), this.state.sort );
```

Comment:
That is too bad you have to wait for the component to mount to start the network request. But ya I guess that makes sense.

I was trying to move the pieces that hide and remove items from the DOM to the constructor because it seemed to speed up. Also we don't load the app until the DOM is rendered so it should be ok to start removing or changing things. Do you see a problem with that? We aren't actually manipulating the DOM for this component.

Reply:
[This comment by a React core dev] explaining why this is the case.

> The specific recommendation to avoid side effects in the constructor [...] is related to the changes we are making to allow rendering to be asynchronous and interruptible (in part to support use cases like this better).

Reply:
As for the DOM manipulation, I'm stumped given that we're manipulating parts of the DOM outside of the React component tree. I'm cautiously leaning toward keeping it inside the constructor if we can quantify it actually speeds up the user experience.

Reply:
It is definitely hard to quantify. Thanks for that link. Very helpful.